crates/re_renderer/src/resource_managers/texture_manager.rs

```diff
@@ -251,12 +268,13 @@ impl TextureManager2D {
     /// Creates a new 2D texture resource and schedules data upload to the GPU if a texture
     /// wasn't already created using the same key.
     pub fn get_or_try_create_with<'a, Err: std::fmt::Display>(
-        &mut self,
+        &self,
         key: u64,
-        texture_pool: &mut GpuTexturePool,
+        texture_pool: &GpuTexturePool,
         try_create_texture_desc: impl FnOnce() -> Result<Texture2DCreationDesc<'a>, Err>,
     ) -> Result<GpuTexture2D, TextureManager2DError<Err>> {
-        let texture_handle = match self.texture_cache.entry(key) {
+        let mut inner = self.inner.lock();
```

Comment:
This _could_ be improved to be a read/write lock where we take the write lock only if a new texture actually needs to be added.

Reply:
if it needs to be added or inserted in `accessed_textures`… it gets a bit hairy, so I opted to keep it simple instead